include/seqan3/alphabet/detail/hash.hpp

```diff
@@ -45,7 +46,7 @@ struct hash<alphabet_t>
  */
 template <ranges::InputRange urng_t>
     //!\cond
-    requires seqan3::Semialphabet<seqan3::reference_t<urng_t>>
+    requires seqan3::Semialphabet<seqan3::reference_t<seqan3::remove_cvref_t<urng_t>>>
```

Comment:
only remove const and remove around the reference_t, not inside. There are non-const views that still have const-reference type, like std::basic_string_view.